core/src/Pos/Aeson/Core/Configuration.hs

```diff
@@ -6,9 +6,26 @@ module Pos.Aeson.Core.Configuration
     (
     ) where
 
-import           Data.Aeson.TH (deriveJSON)
+import           Universum
+
+import           Data.Aeson (FromJSON, parseJSON, withObject, (.:), (.:?))
+import           Data.Aeson.TH (deriveToJSON)
 import           Serokell.Aeson.Options (defaultOptions)
 
 import           Pos.Core.Configuration.Core (CoreConfiguration (..))
+import           Pos.Crypto (RequiresNetworkMagic (..))
+
+deriveToJSON defaultOptions ''CoreConfiguration
 
-deriveJSON defaultOptions ''CoreConfiguration
+instance FromJSON CoreConfiguration where
+    parseJSON = withObject "core" $ \obj -> do
+        ccg   <- obj .: "genesis"
+        ccdsv <- obj .: "dbSerializeVersion"
+        ccrnm <- determineRNM <$> obj .:? "requiresNetworkMagic"
```

Comment:
I *think* this line can be replaced with ([docs](https://hackage.haskell.org/package/aeson-1.4.1.0/docs/Data-Aeson.html#v:.-33--61-))

```
ccrnm <- determineRNM <$> (obj .:? "requiresNetworkMagic" .!= NMMustBeJust)
```

whether that is more or less readable is another question 🙃

Reply:
It would be neater to use `.!=` as you suggest, and then add a comment to highlight the subtlety.
```
-- If "requiresNetworkMagic" is not specified, default to NMMustBeJust
ccrnm <- obj .:? "requiresNetworkMagic" .!= NMMustBeJust
```

Reply:
@mhuesch @rvl Thanks, I wasn't aware of this.